go/pio/host/allwinner/allwinner.go

```diff
@@ -908,19 +1088,40 @@ var mapping = [116][5]string{
 // Defaults to 0x01C20800 as per datasheet if could query the file system.
 func getBaseAddressPB() uint64 {
 	base := uint64(0x01C20800)
+
+	// try the way it looks on pine64 first (Debian ??)
 	link, err := os.Readlink("/sys/bus/platform/drivers/sun50i-pinctrl/driver")
-	if err != nil {
-		return base
-	}
-	parts := strings.SplitN(path.Base(link), ".", 2)
-	if len(parts) != 2 {
-		return base
+	if err == nil {
+		parts := strings.SplitN(path.Base(link), ".", 2)
+		if len(parts) != 2 {
+			return base
+		}
+		base2, err := strconv.ParseUint(parts[0], 16, 64)
+		if err != nil {
+			return base
+		}
+		return base2
 	}
-	base2, err := strconv.ParseUint(parts[0], 16, 64)
-	if err != nil {
-		return base
+
+	// next try the way it looks on C.H.I.P. (Debian wheezy)
+	// note that CHIP's r8 is a repackaging of the a13
+	items, _ := ioutil.ReadDir("/sys/bus/platform/drivers/sun5i-a13-pinctrl/")
```

Comment:
I agree looking at loaded modules is the best check, the functions in internal should be updated.